packages/frontend/src/hooks/useLatestKPITokens.ts

```diff
@@ -12,44 +13,46 @@ export function useLatestKPITokens(limit = 5): {
     const { chain } = useNetwork();
     const provider = useProvider();
 
-    const [kpiTokens, setKPITokens] = useState<KPIToken[]>([]);
-    const [loading, setLoading] = useState(true);
-
-    useEffect(() => {
-        let cancelled = false;
-        async function fetchData(): Promise<void> {
-            if (!chain) return;
-            if (!cancelled) setLoading(true);
-            try {
-                const kpiTokensAmount = await Fetcher.fetchKPITokensAmount({
+    const { data: kpiTokensAmountData, isSuccess: isKPITokensAmountSuccess } =
```

Comment:
Isn't it better to have all in one `queryFn`? This is an atomic operation and I don't see where the single cached items would be used elsewhere.

Reply:
What do you mean by having all in on `queryFn`?

Reply:
This were 3 requests already (dependent on each other):
`Fetcher.fetchKPITokensAmount`
`Fetcher.fetchKPITokenAddresses`
`Fetcher.fetchKPITokens`
 
Just replaced all with react query, I think the code is simpler.

Reply:
Yeah but it doesn't make sense for those 3 requests to have each a cache entry since they act as one single bigger request. That's why I'd just have one cache entry for the latest x KPI tokens, not 3 different entries one for the amount, one for the addresses and one for the tokens themselves.

Reply:
So you're suggesting to only add the last query `Fetcher.fetchKPITokens` to react query, and do the others as before?

Reply:
Nono, to add all of them. Basically just take the `fetchData` function that was used in the hook and pass it as is (with changes due to the fact that you can't use `setState`) to the `queryFn` to React Query.

Reply:
I don't think "useQuery" can be called inside a `useEffect`.

Reply:
but why it's bad to have the 3 queries with react query? what I'm I missing?

I think the code looks clear as well.

Reply:
updated on 59dc180017597dd9a9a1c4b89013aae1e566eab4

Reply:
> but why it's bad to have the 3 queries with react query? what I'm I missing?
> 
> Is less efficient by catching 3 times?

Yeah, pretty sure it's less efficient to have 3 responses cached when the 3 requests as a whole act like an atomic request overall.